bblocks/import_tools/hdr.py

```diff
@@ -0,0 +1,272 @@
+    To get the data as a dataframe, call `get_data` method. You can force an update by calling `update_data`
+    """
+
+    def load_data(self, indicators: str | list = "all") -> ImportData:
+        """Load data to the object
+
+        When called this method checks if the data has been downloaded, if not it downloads it.
+        It then loads the data into the object for selected indicators and returns the object.
+
+        Args:
+            indicators: Indicator(s) to load. If "all", load all indicators.
+
+        Returns:
+            same object with loaded indicators to allow chaining
+        """
+
+        # download data if it does not exist
+        path = BBPaths.raw_data / "HDR.csv"
+        if not path.exists():
+            get_hdr_data().to_csv(path, index=False)
+
+        # load entire dataset to _raw_data if it does not exist
+        if self._raw_data is None:
+            self._raw_data = get_hdr_data()
+
+        # if indicators is "all", load all indicators
+        if indicators == "all":
+            indicators = available_indicators()
+        # if indicators is a string, convert to list
+        if isinstance(indicators, str):
+            indicators = [indicators]
+
+        # check if indicators are valid
+        for indicator in indicators:
+            if indicator not in available_indicators():
+                raise ValueError(f"Indicator {indicator} not found.")
+
+        # load data
+        self._data.update(
+            {
+                indicator: (
+                    self._raw_data[self._raw_data["variable"] == indicator].reset_index(
+                        drop=True
+                    )
+                )
+                for indicator in indicators
+            }
+        )
+        logger.info(f"Data loaded successfully.")
+        return self
+
+    def update_data(self, reload_data: bool = True) -> ImportData:
+        """Update the data
+
+        When called it will update the data used to load indicators.
+
+        Args:
+            reload_data: If True, reload the indicators after updating the data. Default is True.
+
+        Returns:
+            same object with updated data to allow chaining
+        """
+
+        if len(self._data) == 0:
+            raise RuntimeError("No indicators loaded")
+
+        self._raw_data = get_hdr_data()  # update data store in _raw_data attribute
+        self._raw_data.to_csv(BBPaths.raw_data / "HDR.csv", index=False)
+        if reload_data:
+            self.load_data(indicators=list(self._data.keys()))
+
+        logger.info("Data updated successfully.")
+        return self
+
+    def get_data(self, indicators: str | list = "all", **kwargs) -> pd.DataFrame:
+        """Get the data as a dataframe
+
+        Args:
+            indicators: Indicator(s) to get data for. If "all", get all indicators.
+
+        Returns:
+            Loaded indicators as a dataframe
+        """
+
+        return super().get_data(indicators=indicators)
```

Comment:
Since this is the exact `super()` implementation you don't need to reimplement/redefine here. So since this is not an abstract method on the parent class, it will be implemented for all children automatically.

Reply:
You can therefore remove it from this class (also one less thing to test)